### DISCUSSION TITLE
How to disable swipeEnabled screenOption programmatically?

### DISCUSSION THREAD

**Discussion by pavelustenko** [react-navigation/react-navigation#11804] [Q&A]
Hello! I'm using Drawer navigator for it's side panel functionality. I'm using custom drawerContent component but it seems like it's impossible to use swipeable components (such as Slider) in it as Drawer intercepts any right-to-left swipe to close. Is there any way to disable drawer's swipeability programmatically when other swipeable content is present or to disable swipe-to-close functionality without disabling swipe-to-open functionality? I tried to use 
```
this.props.navigation.setOptions({swipeEnabled: false}) 
```
but no luck.
Thanks in advance.

**Comment by svpease** [comment]
I've run into this issue with a competing gesture handler and will have to address this as well eventually..

Even though you can't address it programmatically like you suggest, you could create a React Context that force updates the Navigation's props. Then to avoid all its child components getting re-rendered, wrap them in a useMemo() hook (if that's necessary).
